File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第2章矩形拼接.py
#矩形拼接 2022年第十三届省赛Java大学C组，Python大学A组、C组、研究生组
#lanqiaoOJ 2238


def check1(x1,x2,x3):
    if x1>=x2 and x1>=x3:
        if x1==x2+x3 and a[2]+a[3]-x2==a[4]+a[5]-x3:  return True
    if x2>=x1 and x2>=x3:
        if x2==x1+x3 and a[0]+a[1]-x1==a[4]+a[5]-x3:  return True
    if x3>=x1 and x3>=x2:
        if x3==x1+x2 and a[0]+a[1]-x1==a[2]+a[3]-x2:  return True
    return False
def check2(x1,x2,x3):
    if x1>=x2 and x1>=x3:
        if x1==x2+x3:    return True
    if x2>=x1 and x2>=x3:
        if x2==x1+x3:    return True
    if x3>=x1 and x3>=x2:
        if x3==x1+x2:    return True
    return False
     
T=int(input())
for t in range(T):
    a=list(map(int,input().split()))
    ans=8
    for i in range(0,2):              #第1个矩形
        for j in range(2,4):          #第2个矩形
            for k in range(4,6):      #第3个矩形
                x1,x2,x3 = a[i],a[j],a[k]
                if x1==x2 and x2==x3:          ans = min(ans,4)
                if check1(x1,x2,x3):           ans = min(ans,4)
                if x1==x2 or x1==x3 or x2==x3: ans = min(ans,6)
                if check2(x1,x2,x3):           ans = min(ans,6)
    print(ans)

File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第2章矩形拼接.cpp
//矩形拼接 2022年第十三届省赛Java大学C组，Python大学A组、C组、研究生组
//lanqiaoOJ 2238

//User: root [fzl]的代码

#include<bits/stdc++.h>
using namespace std;
int a[3][2];
int main(){
    int T;    cin >> T;
    while(T--)    {
        for(int i = 0; i < 3; i++)
            cin >> a[i][0] >> a[i][1];
        int ans = 8;
        for(int i = 0; i < 3; i++)               //第1个
            for(int j = 0; j < 3; j++)
                if(i != j)                       //第2个
                    for(int k = 0; k < 3; k++)
                        if(k != i && k != j)     //第3个
                            for(int ii = 0; ii <= 1; ii++){ //第1个有横竖两种摆法
                                for(int jj = 0; jj <= 1; jj++){ //第2个横竖摆
                                    for(int kk = 0; kk <= 1; kk++){ //第3个横竖摆
                                        if(a[i][ii] == a[j][jj] + a[k][kk]){
                                            ans = min(ans, 6);
                                            if(a[j][1-jj] == a[k][1-kk])
                                                ans = min(ans, 4);
                                        }
                                        if(a[i][ii] == a[j][jj] || a[j][jj] == a[k][kk])
                                            ans = min(ans, 6);
                                        if(a[i][ii] == a[j][jj] && a[j][jj] == a[k][kk])
                                            ans = min(ans, 4);
                                    }
                                }
                            }
        cout<<ans<<endl;
    }
    return 0;
}
